packages/ppx/src/declarations_to_string.re

```diff
@@ -0,0 +1,425 @@
+  );
+let inverted_colors =
+  apply(
+    Parser.property_media_inverted_colors,
+    [%expr "inverted-colors"],
+    fun
+    | `Inverted => [%expr "Inverted"]
+    | `None => [%expr "none"],
+  );
+let pointer =
+  apply(
+    Parser.property_media_pointer,
+    [%expr "pointer"],
+    fun
+    | `Coarse => [%expr "coarse"]
+    | `Fine => [%expr "fine"]
+    | `None => [%expr "none"],
+  );
+let hover =
+  apply(
+    Parser.property_media_hover,
+    [%expr "hover"],
+    fun
+    | `Hover => [%expr "hover"]
+    | `None => [%expr "none"],
+  );
+let any_pointer =
+  apply(
+    Parser.property_media_any_pointer,
+    [%expr "any_pointer"],
+    fun
+    | `Coarse => [%expr "coarse"]
+    | `Fine => [%expr "fine"]
+    | `None => [%expr "none"],
+  );
+let any_hover =
+  apply(
+    Parser.property_media_any_hover,
+    [%expr "any_hover"],
+    fun
+    | `Hover => [%expr "hover"]
+    | `None => [%expr "none"],
+  );
+let scripting =
+  apply(
+    Parser.property_media_scripting,
+    [%expr "scripting"],
+    fun
+    | `Enabled => [%expr "enabled"]
+    | `Initial_only => [%expr "initial-only"]
+    | `None => [%expr "none"],
+  );
+
+let resolution =
+  apply(
+    Parser.property_media_resolution,
+    [%expr "scripting"],
+    fun
+    | `Dpcm(v) => render_number(v, "dpcm")
+    | `Dpi(v) => render_number(v, "dpi")
+    | `Dppx(v) => render_number(v, "dppx"),
+  );
+
+let color_index =
+  apply(
+    Parser.property_media_color_index,
+    [%expr "color-index"],
+    render_integer,
+  );
+
+let properties = [
+  ("width", found(width)),
+  ("min-width", found(min_width)),
+  ("max-width", found(max_width)),
+  ("height", found(height)),
+  ("aspect-ratio", found(aspect_ratio)),
+  ("orientation", found(orientation)),
+  ("resolution", found(resolution)),
+  ("grid", found(grid)),
+  ("update", found(update)),
+  ("overflow-block", found(overflow_block)),
+  ("overflow-inline", found(overflow_inline)),
+  ("color", found(color)),
+  ("color-gamut", found(color_gamut)),
+  ("color-index", found(color_index)),
+  ("display-mode", found(display_mode)), // display-mode
+  ("monochrome", found(monochrome)),
+  ("inverted-colors", found(inverted_colors)),
+  ("pointer", found(pointer)),
+  ("hover", found(hover)),
+  ("any-pointer", found(any_pointer)),
+  ("any-hover", found(any_hover)),
+  ("scripting", found(scripting)),
+];
+
+let render_when_unsupported_features = (property, value) => {
+  let to_camel_case = txt =>
+    (
+      switch (String.split_on_char('-', txt)) {
+      | [first, ...remaining] => [
+          first,
+          ...List.map(String.capitalize_ascii, remaining),
+        ]
+      | [] => []
+      }
+    )
+    |> String.concat("");
+
+  /* Transform property name to camelCase since we bind emotion to the Object API */
+  let propertyName = property |> to_camel_case |> render_string;
+  let value = value |> render_string;
+
+  %expr
+  "(" ++ [%e propertyName] ++ ": " ++ [%e value] ++ ")";
+};
```

Comment:
Do we need the rendering unsafe here? I think we handle all cases nicely

Reply:
There are some properties that weren't supported, like some length units, but since this case is just strings and this function only escape the string value, i think its reasonable to handle everything and remove this.